libs/ui/src/components/activity-section-list/activity-section-list.tsx

```diff
@@ -36,67 +35,110 @@ interface Props {
+
+  const sectionListRef = useRef<SectionList<ActivityData, SectionListActivityData>>(null);
 
   const activity = useMemo(() => getFilteredActivity(allActivity, filterTypeName), [allActivity, filterTypeName]);
 
+  const activityDataLength = useMemo(() => {
+    let sum = 0;
+
+    if (sum <= DEFAULT_VISIBILITY_VALUE) {
+      for (const { data } of activity) {
+        sum += data.length;
+      }
+    }
+
+    return sum;
+  }, [activity]);
+
   useEffect(() => {
     if (
       filterTypeName !== ActivityFilterEnum.AllActivity &&
       allActivity.length > 0 &&
-      activity.length === 0 &&
+      activityDataLength <= DEFAULT_VISIBILITY_VALUE &&
       numberOfAttempts <= MAX_ACTIVITY_ATTEMPTS
     ) {
       const lastDate = allActivity.slice(-1)[0].data.slice(-1)[0].timestamp;
 
       fetch(lastDate);
       numberOfAttempts += 1;
     }
-  }, [filterTypeName, allActivity, activity]);
+  }, [filterTypeName, allActivity, activityDataLength]);
 
-  const handleFetchData = () => {
+  const handleFetchData = useCallback(() => {
     if (offsetY === 0) {
       fetch(0);
       numberOfAttempts = 0;
     }
-  };
+  }, [offsetY]);
 
-  useTimerEffect(handleFetchData, DATA_UPDATE_TIME, [publicKeyHash, chainId, offsetY]);
+  useTimerEffect(handleFetchData, DATA_UPDATE_TIME, [publicKeyHash, chainId, offsetY, filterTypeName]);
 
-  const handleScrollWeb = debounce((event: NativeSyntheticEvent<NativeScrollEvent>) => {
-    setOffsetY(event.nativeEvent.contentOffset.y);
-  }, DEBOUNCE_TIME);
+  const debounceContentOffset = debounce(
+    (event: NativeSyntheticEvent<NativeScrollEvent>) => setOffsetY(event.nativeEvent.contentOffset.y),
+    DEBOUNCE_TIME
+  );
 
-  const handleScrollMobile = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
-    event.persist();
-    debounce(() => setOffsetY(event.nativeEvent.contentOffset.y), DEBOUNCE_TIME);
-  };
+  const handleScroll = useCallback((event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    if (isMobile) {
+      event.persist();
+    }
+    debounceContentOffset(event);
+  }, []);
 
-  const handleEndReached = () => {
-    fetch();
-  };
+  const handleEndReached = async ({ distanceFromEnd }: { distanceFromEnd: number }) => {
+    if (distanceFromEnd >= 0) {
+      fetch();
+    }
 
-  const renderItem: SectionListRenderItem<ActivityData, SectionListActivityData> = useCallback(
-    ({ item }) => <ActivityItem transaction={item} address={publicKeyHash} chainName={getDebankId(chainId)} />,
-    [publicKeyHash, chainId]
-  );
+    await sleep(100);
+
+    if (activity.length && distanceFromEnd < 1 && distanceFromEnd >= 0 && sectionListRef.current !== null) {
+      sectionListRef.current?.scrollToLocation({
+        animated: true,
+        itemIndex:
+          activity[activity.length - 1].data.length - 1 === 0 ? 1 : activity[activity.length - 1].data.length - 1,
+        sectionIndex: activity.length - 1,
+        viewOffset: -300
```

Comment:
mb move to constant with meaningful name?

Reply:
I thought about this, but I get that `view Offset` is also a name like a separate constant